src/Microsoft.OData.Edm/Validation/ValidationRules.cs

```diff
@@ -1192,6 +1192,37 @@ public static class ValidationRules
                     }
                 });
 
+        // Add this to rule set as a breaking change for a future release.
+        /// <summary>
+        /// Validates that if a complex-typed property has a type or base type that is the same as

```

Comment:
Is this something required by the standard, or is this something we are adding to help customers since this is likely not what they want?

Reply:
This rule is meant to be a feature to help customers identify issues with recursive structures earlier on in designing their models. It would be a breaking change to add it to the rule set now, but I believe it was discussed that we're planning to add it in the future.